src/main/java/com/ecmsp/orderservice/order/adapter/generator/FixedOrderIdGenerator.java

```
@@ -0,0 +1,21 @@
+package com.ecmsp.orderservice.order.adapter.generator;
+
+import com.ecmsp.orderservice.order.domain.CorrelationId;
+import com.ecmsp.orderservice.order.domain.OrderId;
+import com.ecmsp.orderservice.order.domain.OrderIdGenerator;
+
+import java.util.UUID;
+
+class FixedOrderIdGenerator implements OrderIdGenerator {
+
+
+    @Override
+    public OrderId generate(CorrelationId correlationId) {
+        if (correlationId == null || correlationId.value() == null) {
```

Minor: to be ideal we should have mapping between OrderId and CorrelationId that can be externally set (e.g. in tests) via dedicated API. Therefore, we do not hit the problem where we pass the same CorrelationId and we get conflict on order creation with the same ID. It also provide looser coupling between context and order.

---

Apart from it, it might be better to provide fallback to random OrderId generation if there is no mapping set.

---

see if you like it XD